unitest for iip15-manager contract

millken opened the pull request:
# Description
tests case for iip15manager contract

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

Liuhaai commented on action/protocol/execution/testdata-istanbul/iip15-manager.sol:
```diff
@@ -0,0 +1,43 @@
+        return isApproved[_contractAddress];
+    }
+
+}
```
new line

Liuhaai commented on action/protocol/execution/testdata-istanbul/iip15-manager.json:
```diff
@@ -0,0 +1,254 @@
+        "comment": "approveContract use new owner, success"
+    }
+]
+}
```
same

ququzone commented on action/protocol/execution/testdata-istanbul/iip15-manager.sol:
```diff
@@ -0,0 +1,43 @@
+
+contract IIP15Manager is Ownable {
+    mapping (address => address) private contractMapping;
+    mapping (address => bool) private isApproved;
```
Maybe merge contractMapping and isApproved into struct is prefect.

ququzone commented:
By the way, does it need `deregister` function? If want to add extra APIs in the feature then need to deploy this contract as upgradeable.

millken commented:
@ququzone @dustinxie update contract, use map => struct to store contract info, added `getContract`, `removeContract` APIs, deleted `isApproved`,`isRegistered` API(use getContract instead of, only call once)

ququzone commented on action/protocol/execution/testdata-istanbul/iip15-manager.sol:
```diff
@@ -0,0 +1,56 @@
+
+    struct ContractInfo {
+        address recipient;
+        bool isRegistered;
```
This field is unnecessary and you can use recipient is address(0) to check whether registered.
